contracts/src/contracts/ACP99/ACP99Manager.sol

```diff
@@ -351,31 +357,56 @@ contract ACP99Manager is Ownable2Step, IACP99Manager {
 
         if (weight == 0) {
             if (nonce != (validation.periods.length)) {
-                revert ACP99Manager__InvalidSetSubnetValidatorWeightNonce(
+                revert ACP99Manager__InvalidSetL1ValidatorWeightNonce(
                     nonce, uint64(validation.periods.length)
                 );
             }
             validation.status = ValidationStatus.Completed;
-            subnetTotalWeight -= validation.periods[nonce - 1].weight;
+            l1TotalWeight -= validation.periods[nonce - 1].weight;
         } else {
             if (nonce != (validation.periods.length - 1)) {
-                revert ACP99Manager__InvalidSetSubnetValidatorWeightNonce(
+                revert ACP99Manager__InvalidSetL1ValidatorWeightNonce(
                     nonce, uint64(validation.periods.length - 1)
                 );
             }
             validation.status = ValidationStatus.Active;
             validation.periods[nonce].startTime = uint64(block.timestamp);
-            subnetTotalWeight +=
-                validation.periods[nonce].weight - validation.periods[nonce - 1].weight;
+            l1TotalWeight += validation.periods[nonce].weight - validation.periods[nonce - 1].weight;
         }
 
         _notifySecurityModuleValidatorWeightUpdate(validationID, nonce, weight);
 
         emit CompleteValidatorWeightUpdate(validation.nodeID, validationID, nonce, weight);
     }
 
+    function _validatePChainOwner(
```

Comment:
Since we add this function, should we use it in `initiateValidatorRegistration`?

Reply:
Like [here](https://github.com/ava-labs/teleporter/blob/fa9d24a7fbb388e6316f19e9c78b44dbf12cd772/contracts/validator-manager/ValidatorManager.sol#L249)

Reply:
Good catch